pkg/rdt/rdt.go

```diff
@@ -294,6 +294,12 @@ func GetMonFeatures() map[MonResource][]string {
 	return map[MonResource][]string{}
 }
 
+// IsQualifiedClassName returns true if given string qualifies as a class name
+func IsQualifiedClassName(name string) bool {
+	// Must be qualified as a file name
+	return name != "." && name != ".." && !strings.ContainsRune(name, '/')
```

Comment:
maybe there are better functions to validate names? because there can be bunch of other non-numeric symbols that will become invalid. (\t\n as most obvious ones). I think in k8s code there were examples of validation that will check that it starts with alphanumeric, then continues with alphanumerics/dots/underscores and not longer than X symbols... Something like that.

Reply:
> maybe there are better functions to validate names? because there can be bunch of other non-numeric symbols that will become invalid. (\t\n as most obvious ones).

I don't think we should try to be overly smart here. Directory names like `"foo\nbar\t"` are valid. I agree that it is a super exotic corner case, but, nevertheless. Length checking could be added, limit to 4095 or so.

Reply:
well, that's one of the basics of secure argument handling. This parameters comes from user unformatted annotation. Good sanity check on it is needed.

Reply:
> Directory names like `"foo\nbar\t"` are valid.

well, actually, based on kernel code, it is not valid: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/arch/x86/kernel/cpu/resctrl/rdtgroup.c?h=v5.12.3#n2978

Reply:
> well, that's one of the basics of secure argument handling. This parameters comes from user unformatted annotation. Good sanity check on it is needed.

Well, yes, but we should allow whatever is accepted on the (pseudo-)filesystem level.

Reply:
> well, actually, based on kernel code, it is not valid:

Good catch! I'll add that